tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -805,6 +810,370 @@ def test_csrlsvqr(self, dtype):
+        """
+        A = self._generate_A_for_fiedler(12, 2, xp)
+        X = self._generate_approximate_X_for_fiedler(12, 2, xp)
+        lobpcg_w, lobpcg_V = sp.linalg.lobpcg(A, X, largest=False)
+        return lobpcg_w, _eigen_vec_transform(lobpcg_V, xp)
+
+    def _verbosity_helper(self, xp, sp):
+        """Helper to capture the verbose output from stdout
+        """
+        A, B = self._generate_input_for_elastic_rod(100, xp)
+        n = A.shape[0]
+        m = 20
+        X = self._generate_random_initial_ortho_eigvec(n, m, xp)
+        saved_stdout = io.StringIO()
+        with contextlib.redirect_stdout(saved_stdout):
+            _, _ = sp.linalg.lobpcg(A, X, B=B, tol=1e-5,
+                                    maxiter=30, largest=False,
+                                    verbosityLevel=9)
+        output = saved_stdout.getvalue().strip()
+        return output
+
+    def test_verbosity(self):
+        """Check that nonzero verbosity level code runs
+           and is identical to scipy's output format.
+        """
+        stdout_cupy = self._verbosity_helper(cupy, cupyx.scipy.sparse)
+        stdout_numpy = self._verbosity_helper(numpy, scipy.sparse)
+        # getting rid of the numbers and whitespaces, we care only about
+        # format of printed output.
+        # also, due to the fact that there are unpredictable (but minor)
+        # differences in decimal digits between scipy and cupy verbose output
+        stdout_cupy = re.sub(r'[-+]?\d+\.?\d*[ ]*', '{number}', stdout_cupy)
+        stdout_numpy = re.sub(r'[-+]?\d+\.?\d*[ ]*', '{number}', stdout_numpy)
+        assert stdout_numpy == stdout_cupy, '''numpy: %s
+                                               cupy: %s''' % (stdout_numpy,
+                                                              stdout_cupy)
+
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-3, sp_name='sp',
+                                 contiguous_check=False)
+    def test_random_initial_float32(self, xp, sp):
+        """Check lobpcg in float32 for specific initial.
+        """
+        n = 50
+        m = 4
+        vals = -xp.arange(1, n + 1)
+        A = sp.diags([vals], [0], (n, n))
+        A = A.astype(xp.float32)
+        X = testing.shaped_random((n, m), xp=xp, seed=3)
+        eigvals, vecs = sp.linalg.lobpcg(A, X, tol=1e-3, maxiter=50,
+                                         verbosityLevel=1)
+        return eigvals, _eigen_vec_transform(vecs, xp)
+
+    def test_maxit_None(self):
+        """Check lobpcg if maxit=None runs 20 iterations (the default)
+        by checking the size of the iteration history output, which should
+        be the number of iterations plus 2 (initial and final values).
+        """
+        n = 50
+        m = 4
+        vals = -cupy.arange(1, n + 1)
+        A = sparse.diags([vals], [0], (n, n))
+        A = A.astype(cupy.float32)
+        X = testing.shaped_random((n, m), xp=cupy, seed=1566950023)
```

Comment:
Is this seed randomly typed one? `len(l_h)` is `22` in most seed value?

Reply:
in cuda11.0 it seems to work irrespective of seed value